app/src/main/java/com/example/taskyy/ui/events/LoginEvents.kt

```diff
@@ -0,0 +1,9 @@
+package com.example.taskyy.ui.events
+
+class LoginEvents {
+    companion object {
+        fun textChanged(): LoginEvents {
+            return LoginEvents()
+        }
+    }
+}
```

Comment:
This class doesn't make much sense to me, usually you implement the UI-to-ViewModel events like this:
```
sealed interface LoginEvent {
    data class OnEmailChanged(val email: String): LoginEvent
    data object OnLoginClick: LoginEvent
    ...
}
```
And in your ViewModel:
```
fun onEvent(event: LoginEvent) {
    when(event) {
        is LoginEvent.OnEmailChanged -> state = state.copy(email = event.email)
        is LoginEvent.Login -> login()
        ...
    }
}
```

Reply:
Why use 'data' here? Also why use object for the click but class for the email change? Is it because you're updating the state value?

Reply:
Data classes and objects just come with already implemented `equals`, `hashCode()` and `copy()` functions which is very convenient.
Normally, we'd expect that these are the same:
```
val class1 = MyClass("HelloWorld")
val class2 = MyClass("HelloWorld")
println(class1 == class")
```
But this will only print true if `MyClass` is a data class, since it will then compare their values rather than whether they're the same reference (they're two different instances, so with a normal class the comparison would return false)